### ISSUE TITLE
Release 5.1.0 Breaking Change

### ISSUE THREAD

**Issue by bensmind** [khellang/Scrutor#244]
With the introduction of the breaking change in #238. AddClasses can no longer be used to bind internal implementations to public interfaces. This change doesn't seem to makes sense given the overload that accepts the `publicOnly` flag. To maintain higher flexibility and not to risk breaking changes.

I think #238 should be reverted and the docs should be updated to be consistent with the existing behavior. 
- [Relevant Interface/Documentation](https://github.com/khellang/Scrutor/blob/df99e1702db14b07203a691e242ecc4d024a418c/src/Scrutor/IImplementationTypeSelector.cs#L15)

If the `publicOnly` overloads did not exist I think a better case could be made to change the implementation, but given that they do, that is the mechanism that should be used to support both scenarios.

**Comment by khellang** [comment]
> AddClasses can no longer be used to bind internal implementations to public interfaces

This is not true, you just need to pass `publicOnly: false` and then you're back to where it was 😄

**Comment by ondrejkretek-oriflame** [comment]
But still it is a breaking change introduced in a minor version. Our application broke in runtime after update to 5.1.0 because internal classes stopped to be added to the container.

**Comment by khellang** [comment]
Alright, I've unlisted v5.1.0 and v5.1.1 from NuGet.

I've also released the following two versions:
- v5.1.2 that's exactly the same as v5.1.1, but without the breaking change
- v6.0.0 that is v5.1.1 with the breaking change.

Hopefully that'll make people happy 😅
